version: '3.8'
services:
  pythonapp:
    build: ./python/
    command: ["bash", "-c", "./wait-for-it.sh mysql:3306 --timeout=30 --strict -- flask run --host=0.0.0.0 & python /python/app/iota/Trigger.py main"]
    volumes:
      - ./python:/python
    environment:
      - FLASK_APP=server.py
      - FLASK_ENV=development
    ports:
      - '5000:5000'
    networks:
      - custom_network
    depends_on:
      - mysql

  tests:
    build: ./python/
    command: ["/python/app/wait-for-it.sh", "mysql:3306", "--", "python", "-m", "unittest", "discover"]
    networks:
      - custom_network
    depends_on:
      - mysql

  mysql:
    image: mysql:latest
    environment:
      MYSQL_ROOT_PASSWORD: Root
      MYSQL_DATABASE: DB
      MYSQL_USER: user
      MYSQL_PASSWORD: pass
    volumes:
      - ./mysql/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - custom_network
    ports:
      - "3306:3306"
    restart: unless-stopped

volumes:
  mysql_data:

networks:
  custom_network: